api/common/interfaces.go

```diff
@@ -953,6 +958,75 @@ func findSuccessorAddr(s Serverer, params map[string]interface{}, ctx context.Co
 	return respPacking(errcode.SUCCESS, addrs[0])
 }
 
+// getPeerAddr get a node address
+// params: {"address":<address>}
+// return: {"resultOrData":<result>|<error data>, "error":<errcode>}
+func getPeerAddr(s Serverer, params map[string]interface{}, ctx context.Context) map[string]interface{} {
+	if len(params) < 1 {
+		return RespPacking("length of params is less than 1", errcode.INVALID_PARAMS)
+	}
+
+	str, ok := params["address"].(string)
+	if !ok {
+		return RespPacking("address should be a string", errcode.INTERNAL_ERROR)
+	}
+
+	clientID, _, _, err := address.ParseClientAddress(str)
+	if err != nil {
+		return RespPacking(err.Error(), errcode.INTERNAL_ERROR)
+	}
+
+	wsAddr, rpcAddr, pubkey, id, err := s.GetNetNode().FindWsAddr(clientID)
+	if err != nil {
+		return RespPacking(err.Error(), errcode.INTERNAL_ERROR)
+	}
+
+	n := s.GetNetNode()
+	if n == nil {
+		return nil
+	}
+
+	if n.GetWsAddr() == wsAddr {
+		offer := params["offer"].(string)
+		peer := webrtc.NewPeer(config.Parameters.StunList)
+
+		err = peer.Answer(offer)
+		if err != nil {
+			return RespPacking(err.Error(), errcode.INTERNAL_ERROR)
+		}
+		answer := <-peer.OnSdp
```

Comment:
need a timeout to prevent goroutine halt

Reply:
Yes. timeout is added.